### PR TITLE
Added bip and private key utilities

### PR DESCRIPTION
**Pull request by edtubbs**
Exposes bip32, bip44 and private key encoding utilities.

### CODE REVIEW

**Review by xanimo — APPROVED** [review]
ACK, tested on x86_64-pc-linux-gnu.

**Comment by michilumin** [comment]
OK to merge just TODO: later want to change this so it doesn't require a pointer and memory address reference because it cuts off accessibility to some other languages; want to retool it so that a chainparam is selected via an option/integer/value, not feeding it the chainparams pointer.

File: contrib/examples/example.c
Comment (michilumin) [inline contrib/examples/example.c:1]: TODO: want libdogecoin api to eventually never require pointers, so should replace pointer addr input with parameter.

Comment (michilumin) [reply contrib/examples/example.c:1]: TODO: same here with pointer intake, should require main/regtest/testnet as option for chainparams not the actual chainparams structure address

**Review by michilumin — APPROVED** [review]
Approving but with the caveat of later rework on needing to manipulate/send a pointer addr to use.

File: contrib/examples/example.c
Comment (edtubbs) [reply contrib/examples/example.c:1]: Agreed, I'll work to wrap the functions with pointers so that they're not necessary.

**Pull request merged**